managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -255,6 +260,64 @@ public void initializeFailed(ManagedLedgerException e) {
         });
     }
 
+
+
+    @Override
+    public ReadOnlyCursor openReadOnlyCursor(String managedLedgerName, Position startPosition, ManagedLedgerConfig config)
+            throws InterruptedException, ManagedLedgerException {
+        class Result {
+            ReadOnlyCursor c = null;
+            ManagedLedgerException e = null;
+        }
+        final Result r = new Result();
+        final CountDownLatch latch = new CountDownLatch(1);
+        asyncOpenReadOnlyCursor(managedLedgerName, startPosition, config, new OpenReadOnlyCursorCallback() {
+            @Override
+            public void openReadOnlyCursorComplete(ReadOnlyCursor cursor, Object ctx) {
+                r.c = cursor;
+                latch.countDown();
+            }
+
+            @Override
+            public void openReadOnlyCursorFailed(ManagedLedgerException exception, Object ctx) {
+                r.e = exception;
+                latch.countDown();
+            }
+        }, null);
+
+        latch.await();
+
+        if (r.e != null) {
+            throw r.e;
+        }
+        return r.c;
+    }
+
+    @Override
+    public void asyncOpenReadOnlyCursor(String managedLedgerName, Position startPosition, ManagedLedgerConfig config,
+            OpenReadOnlyCursorCallback callback, Object ctx) {
+        checkArgument(startPosition instanceof PositionImpl);
```

Comment:
Sure though, we have that same check in multiple places in the code already. It's more of an assertion since we don't have any other implementations outside of `PositionImpl`